Tests/iaas/flavor-naming/flavor-add-extra-specs.py

```diff
@@ -298,64 +208,77 @@ def main(argv):
                 errors += 1
-                continue
 
-        # Now do sanity checks (std properties)
-        stderrs = check_std_props(flavor, flvnm, " by name")
-        errors += stderrs
-        if stderrs:
-            continue
-
-        # Parse and Generate name-v1 and name-v2
-        upd = check_name_extra(flavor, "v2", not is_v1, flname)
-        if upd:
-            errors += update_flavor_extra(compute, flavor, "scs:name-v2")
-            chg += 1
-        upd = check_name_extra(flavor, "v1", is_v1, flname)
-        if upd:
-            errors += update_flavor_extra(compute, flavor, "scs:name-v1")
-            chg += 1
-        # Parse and Generate cpu-type and disk0-type
-        upd = check_extra_type(flavor, "cpu-type", flvnm.cpuram.cputype, CPUTYPE_KEY)
-        if upd:
-            errors += update_flavor_extra(compute, flavor, "scs:cpu-type")
-            chg += 1
-
-        # We may not have a disk (or the type is unknown)
-        if flvnm.disk:
-            dtp = flvnm.disk.disktype
-            if not dtp:
-                dtp = disk0_type
+        # select a reference flavorname, check flavornames
+        if flavornames:
+            flavorname_items = iter(flavornames.items())
+            reference_key, reference = next(flavorname_items)
+            reference_core = _extract_core(reference)
+            # sanity check: claims must be true wrt actual flavor
+            errors += check_std_props(flavor, reference, " by name")
+            # sanity check: claims must coincide (check remaining flavornames)
+            for key, flavorname in flavorname_items:
+                errors += check_std_props(flavor, flavorname, " by name")
+                core = _extract_core(flavorname)
+                if core != reference_core:
+                    logger.error(f"Inconsistent {key} vs. {reference_key}: {core} vs. {reference_core}")
         else:
-            dtp = None
-        upd = check_extra_type(flavor, "disk0-type", dtp, DISKTYPE_KEY)
-        if upd:
-            errors += update_flavor_extra(compute, flavor, "scs:disk0-type")
+            # we need cputype and disktype from user
+            if not cpu_type:
+                logger.warning(f"Need to specify cpu-type for generating name for {flavor.name}, skipping")
+                continue
+            if flavor.disk and not disk0_type:
+                logger.warning(f"Need to specify disk0-type for generating name for {flavor.name}, skipping")
+                continue
+            reference_key = None
+            reference = generate_flavorname(flavor, cpu_type, disk0_type)
+
+        expected = flavorname_to_dict(reference)
+        # set value for any unexpected, but present key to default value
+        # this will help us find necessary removals
+        for key in flavor.extra_specs:
+            expected.setdefault(key, DEFAULTS.get(key))
+
+        # generate default name-vN (ONLY if none present)
```

Comment:
My suggestion would be to generate the extra names for a full set of 2 to 4 flavor scs:name-vN names.

Reply:
Isn't this what's being done?

Reply:
BTW, you seem to have been reviewing and outdated commit there.